File: LeetCode/problems/p0028/java/src/p028/java/P028.java
//[COMPLETED]

package p028;

public class P028 {

    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(s.strStr("a", "a"));
        System.out.println(s.strStr("hello", "ll"));
        System.out.println(s.strStr("aaaaa", "bba"));
        System.out.println(s.strStr("abcd", ""));
        System.out.println(s.strStr("", "a"));
        System.out.println(s.strStr("", ""));
        
    }
    
    
private static class Solution {
    public int strStr(String haystack, String needle) {
        if (haystack.equals("") && needle.equals(""))
            return 0;
        for (int i = 0; i <= haystack.length() - needle.length(); i++) {
            if (haystack.substring(i, i + needle.length()).equals(needle) )
                return i;
        }
        return -1;
    }
}

}

File: LeetCode/problems/p0028/python/p028.py
#[COMPLETED]

class Solution(object):
    """
    :type haystack: str
    :type needle: str
    :rtype: int
    """
    def strStr(self, haystack, needle):
        if needle == "":
            return 0
        for i in range(len(haystack) - len(needle) + 1):
            if haystack[i:i+len(needle)] == needle:
                return i
        return -1


sol = Solution()
print(sol.strStr("a", "a"), "expected 0")
